### DISCUSSION TITLE
fictitious Boundaries

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#24211] [Q&A Modules: General]
Is it possible to implement fictitious boundary methods in moose, and are there any examples to demonstrate how?

**Comment by GiudGiud** [comment]
Hello

Do you have a link? I dont recall what that is
Which field is this for?

Guillaume

**Reply by jiangwen84** [reply]
@maxnezdyur

**Reply by TLWise** [reply]
Thank you for inquiring. Here is a paper that discusses the method.
https://www.sciencedirect.com/science/article/abs/pii/S0021999100965422

**Reply by hugary1995** [reply]
Yes, @maxnezdyur is developing a MOOSE-based app for that.

**Reply by TLWise** [reply]
Will it be made available for beta testers, in particular me? Is there an existing preliminary framework for simple 1D models?

**Reply by maxnezdyur** [reply]
It's still a work in progress, with a few kinks that need to be taken care of. I want to make it public at the end of the summer. The app is more of an embedded boundary method than the fictitious method you linked, but it should still be helpful, depending on what you are trying to do.

**Reply by TLWise** [reply]
It appears from what you @maxnezdyur are describing, is that implementing FBM can not be easily applied by using existing apps and moose interfaces. An app and custom code has to be created to accomplish it.

**Reply by TLWise** [reply]
@maxnezdyur , If I am implementing a solution method that solves the Poisson problems A*u = f in the Matrix form A_h*U_h =  F_h, where S_h represents the boundaries of the problem domain. Is it possible in MOOSE to modify the total solution Matrix of the problem to ignore the surface boundaries of the problem, or would there have to be a modification of the Petsc code to perform such actions?

**Reply by GiudGiud** [reply]
hello

you can  implement a boundary condition (BC) object. This lets you select the flux value at the boundary
The assembly in the volume is intended to be performed with kernels in MOOSE. 

> Is it possible in MOOSE to modify the total solution Matrix of the problem to ignore the surface boundaries of the problem,

I think you should code in MOOSE what you need at the boundary. Not try to hack the matrix after the fact.
If you just want to hack the matrix, you will have to modify the FEProblemSolve/NonlinearSystem classes in MOOSE. This is not what we intend for MOOSE and we probably would not support you in doing that.
